include/seqcmd.h

```diff
@@ -0,0 +1,510 @@
+                      (u8)(numSeqRequests))
+
+/**
+ * Setup a request to restore a volume on target seqPlayer once a setup seqPlayer is finished playing and disabled.
+ * Specifically, it will only restore volume if the number of queued requests on the setup seqPlayer matches
+ * the number of sequences queued
+ *
+ * @param setupSeqPlayerIndex the index of the seqPlayer to wait for to be disabled
+ * @param targetSeqPlayerIndex the index of the seqPlayer to modify
+ * @param scaleIndex the scale index of a seqPlayer
+ * @param duration duration to transition to the volume
+ */
+#define SEQCMD_SETUP_RESTORE_PLAYER_VOLUME_WITH_SCALE(setupSeqPlayerIndex, targetSeqPlayerIndex, scaleIndex, duration) \
+    Audio_QueueSeqCmd((SEQCMD_OP_SETUP_CMD << 28) | (SEQCMD_SUB_OP_SETUP_RESTORE_VOLUME_WITH_SCALE << 20) |            \
+                      ((u8)(setupSeqPlayerIndex) << 24) | ((u8)(targetSeqPlayerIndex) << 16) |                         \
+                      ((u8)(scaleIndex) << 8) | (u8)(duration))
+
+/**
+ * Setup a request to disable (or reenable) specific channels on a target seqPlayer
+ * once a setup seqPlayer is finished playing and disabled.
+ *
+ * @param setupSeqPlayerIndex the index of the seqPlayer to wait for to be disabled
+ * @param targetSeqPlayerIndex the index of the seqPlayer to modify
+ * @param channelMask a 16 bit mask where each bit maps to a channel. Bitflag on to disable
+ */
+#define SEQCMD_SETUP_SET_CHANNEL_DISABLE_MASK(setupSeqPlayerIndex, targetSeqPlayerIndex, channelMask)      \
+    Audio_QueueSeqCmd((SEQCMD_OP_SETUP_CMD << 28) | (SEQCMD_SUB_OP_SETUP_SET_CHANNEL_DISABLE_MASK << 20) | \
+                      ((u8)(setupSeqPlayerIndex) << 24) | ((u8)(targetSeqPlayerIndex) << 16) | (u16)(channelMask))
+
+/**
+ * Queue a request to scale the frequency of an entire sequence on a target seqPlayer
+ * once a setup seqPlayer is finished playing and disabled.
+ *
+ * @param setupSeqPlayerIndex the index of the seqPlayer to wait for to be disabled
+ * @param targetSeqPlayerIndex the index of the seqPlayer to modify
+ * @param duration duration to transition to the frequency
+ * @param freqScale the scaling factor to shift the pitch. Ranged from 0-0xFFFF, with 100 mapping to 1.0f
+ *
+ * @note The base value for frequency, 100, is 10 times smaller than other fruequency commands.
+ * 200 will double the frequency (raise an octave), 50 will half the frequency (lower an octave).
+ */
+#define SEQCMD_SETUP_SET_PLAYER_FREQ(setupSeqPlayerIndex, targetSeqPlayerIndex, duration, freqScale)                        \
+    Audio_QueueSeqCmd((SEQCMD_OP_SETUP_CMD << 28) | (SEQCMD_SUB_OP_SETUP_SET_PLAYER_FREQ << 20) |                      \
+                      ((u8)(setupSeqPlayerIndex) << 24) | ((u8)(targetSeqPlayerIndex) << 16) | ((u8)(duration) << 8) | \
+                      (u8)(freqScale))
+
+/**
+ * Queue a request to discard audio data by popping one more entries from the persistent caches on the audio heap,
```

Comment:
I noticed this (`SEQCMD_SETUP_POP_PERSISTENT_CACHE`) is unused but the underlying `AudioHeap_PopPersistentCache` isn't, as `AUDIO_CMD_E3` is QueueCmd'ed by the fanfare-playing-thing `func_800F5CF8`

I added prints to `AudioHeap_PopPersistentCache` like so:
```diff
diff --git a/src/code/audio_heap.c b/src/code/audio_heap.c
index 7cdfe086b..683b2302b 100644
--- a/src/code/audio_heap.c
+++ b/src/code/audio_heap.c
@@ -257,3 +257,6 @@ void AudioHeap_PopPersistentCache(s32 tableType) {
 
+    osSyncPrintf("AudioHeap_PopPersistentCache(%d) ", tableType);
+
     if (persistent->numEntries == 0) {
+        osSyncPrintf("numEntries == 0\n");
         return;
@@ -272,2 +275,4 @@ void AudioHeap_PopPersistentCache(s32 tableType) {
 
+    osSyncPrintf("entry.id = %d\n", (s32)persistent->entries[persistent->numEntries - 1].id);
+
     loadStatus[persistent->entries[persistent->numEntries - 1].id] = LOAD_STATUS_NOT_LOADED;
```

and triggered fanfare usage:

```
// 1 warp from title screen to zora's river with map select
// 2 grab deku seeds (plays item get fanfare)
Core: IS64: [mAudioHeap_PopPersistentCache(0) numEntries == 0
Core: IS64: AudioHeap_PopPersistentCache(1) numEntries == 0
// 3 grab magic (plays item get fanfare)
Core: IS64: [mAudioHeap_PopPersistentCache(0) entry.id = 34
Core: IS64: AudioHeap_PopPersistentCache(1) entry.id = 35

// 4 warp to zora's river (reload) with map select
// 5 grab deku seeds (plays item get fanfare)
Core: IS64: [mAudioHeap_PopPersistentCache(0) numEntries == 0
Core: IS64: AudioHeap_PopPersistentCache(1) numEntries == 0
// 6 get skulltula token (plays whatever fanfare)
Core: IS64: AudioHeap_PopPersistentCache(0) entry.id = 34
Core: IS64: AudioHeap_PopPersistentCache(1) entry.id = 35
// 7 buy beans (plays item get fanfare)
Core: IS64: [mAudioHeap_PopPersistentCache(0) entry.id = 57
Core: IS64: AudioHeap_PopPersistentCache(1) entry.id = 35
// 8 play saria's
Core: IS64: AudioHeap_PopPersistentCache(0) entry.id = 34
Core: IS64: AudioHeap_PopPersistentCache(1) entry.id = 35
// 9 play saria's again
Core: IS64: AudioHeap_PopPersistentCache(0) entry.id = 68
Core: IS64: AudioHeap_PopPersistentCache(1) numEntries == 0
```

I would *guess* the idea is to free the previous fanfare's resources but does only the fanfare stuff use that "persistent cache" then, I forgot if you told us anything like that

Reply:
Right, so I realized the reason I put "one more" is because the audio thread command to pop the persistent cache is also called in `Audio_UpdateFanfare` (`func_800F5CF8`) when a fanfare is ready to play, so it clears out the previous contents of the persistent cache. But I suppose it doesn't matter what other code does, so I can just simplify the phrasing to "remove an entry..."